app/assets/javascripts/pageflow/editor/models/uploaded_file.js

```diff
@@ -10,6 +10,13 @@ pageflow.UploadedFile = Backbone.Model.extend({
 
     this.configuration.i18nKey = this.i18nKey;
 
+    if (pageflow.editor.nextUploadTargetFile){
+      this.set({parent_file_id: pageflow.editor.nextUploadTargetFile.get('id'),
+                parent_file_model_type: pageflow.editor.nextUploadTargetFile.fileType().typeName});
+      this.collection = pageflow.editor.nextUploadTargetFile
+        .nestedFiles(this.fileType().collectionName);
+    }
+
```

Comment:
How does a FilesCollection know the parent file other than via `pageflow.editor.nextUploadTargetFile`? We create any nestedFiles collection in this model and do not store direct information about the parent in such a collection